egui/src/widgets/plot/items.rs

```diff
@@ -223,4 +544,13 @@ impl Curve {
         self.name = name.to_string();
         self
     }
+
+    /// Return the color by which the curve can be identified.
+    pub(crate) fn get_color(&self) -> Option<Color32> {
+        self.color.filter(|color| color.a() != 0).or_else(|| {
+            self.marker
+                .map(|marker| marker.color)
+                .filter(|color| *color != Color32::TRANSPARENT)
+        })
```

Comment:
this again makes me think we should only have one color. Likely users will either show markers or a line. If they want to show both, it is likely they will be separate anyway (e.g. show a smooth line and the sparse points they interpolate)

Reply:
Done in ae00424